src/SeoInspector.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+import fs from 'fs';
+import { JSDOM } from 'jsdom';
+import _ from 'lodash';
+import conv from 'iconv-lite';
+import RuleFactory from './RuleFactory';
+
+const defaultRules = {
+    'ImgTagWithAltAttritube': { enabled: 1 },
+    'ATagWithRelAttritube': { enabled: 1 },
+    'HeadTagWithTitleAndDescriptionsKeywordsMeta': { enabled: 1 },
+    'NoTooManyStrongTags': { enabled: 1, threshold: 15 },
+    'NoMoreThanOneH1Tag': { enabled: 1 }
+};
+
+class SeoInspector {
+
+    constructor(options = {}) {
+        this.done = (options.done) ? options.done : (err) => { if (err) throw err };
+        this.input = null;
+        this.output = null;
+        this.rules = {};
+        this.ruleFactory = new RuleFactory();
+        return this;
+    }
+
+    read(input) {
+        this.input = input;
+        return this;
+    }
+
+    addRule(name, options) {
+        if (_.has(defaultRules, name)) {
+            _.set(this.rules, name, options);
+        }
+        return this;
+    }
+
+    write(output) {
+        this.output = output;
+        if (_.isEmpty(this.rules)) {
+            this.rules = defaultRules;
+        }
+        this._run();
+    }
+
+    _run() {
+        this._getHtml()
+        .then(
+            html => this._getDom(html)
+        )
+        .then(
+            dom => this._getSummary(dom)
+        )
+        .then(
+            summary => this._output(summary)
+        );
+    }
+
+    _getHtml() {
+    	return new Promise((resolve, reject) => {
+    	    const type = this.input.constructor.name;
+    	    switch (type) {
+    	        case 'String':
+    	            const filePath = this.input;
+    	            fs.readFile(filePath, 'utf8', (err, html) => {
+    	                if (err) return this.done(err);
+    	                resolve(html);
+    	            });
+    	            break;
+    	        case 'ReadStream':
```

Comment:
What if exception occurs during stream reading ?

Reply:
readStream.on('error', (e) => {
                        this.done(e);
                    });